src/Medidata.ZipkinTracer.Core.Collector/SpanProcessor.cs

```diff
@@ -53,71 +59,73 @@ public virtual void Stop()
 
         public virtual void Start()
         {
-            spanProcessorTaskFactory.CreateAndStart(() => LogSubmittedSpans());
+            spanProcessorTaskFactory.CreateAndStart(LogSubmittedSpans);
         }
 
         internal virtual void LogSubmittedSpans()
         {
             Span span;
-            spanQueue.TryTake(out span);
-            if (span != null)
+            while (spanQueue.TryTake(out span))
             {
-                logEntries.Add(Create(span));
+                serializableSpans.Enqueue(new SerializableSpan(span));
                 subsequentPollCount = 0;
             }
-            else if (logEntries.Any())
-            {
-                subsequentPollCount++;
-            }
+            if(serializableSpans.Count > 0) subsequentPollCount++;
 
-            if ((logEntries.Count() >= maxBatchSize)
-                || (logEntries.Any() && spanProcessorTaskFactory.IsTaskCancelled())
-                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
-            {
-                var entries = logEntries;
-                logEntries = new List<LogEntry>();
-                subsequentPollCount = 0;
-                Log(clientProvider, entries);
-            }
-        }
 
-        internal void Log(IClientProvider client, List<LogEntry> logEntries)
-        {
-            try
-            {
-                clientProvider.Log(logEntries);
-                retries = 0;
-            }
-            catch (TException)
+            if ((serializableSpans.Count() >= maxBatchSize)
```

Comment:
ss? We will send after a second or two though if nothing else is added to the queue.